### PR TITLE
[ROOT master] Integrate new PyROOT with CMSSW

### PR DESCRIPTION
**Pull request by mrodozov**
please test
no updates from https://sft.its.cern.ch/jira/browse/ROOT-10545 but lets get the new commits

### CODE REVIEW

**Comment by mrodozov** [comment]
please test
this should build at least the externals

**Comment by mrodozov** [comment]
test parameters:
- pull_request=https://github.com/cms-sw/cmssw/pull/28933

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
abort

**Comment by mrodozov** [comment]
test parameters:
- pull_request = https://github.com/cms-sw/cmssw/pull/28951

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
test parameters:
- pull_request=https://github.com/cms-sw/cmssw/pull/28965

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test
build the cmssw-tool-conf

**Comment by mrodozov** [comment]
abort

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

File: pip/py2-hepdata-lib.file [hunk 1]
```diff
@@ -0,0 +1 @@
+%define PipPreBuild export CPPYY_BACKEND_LIBRARY=${ROOT_ROOT}/${PYTHON_LIB_SITE_PACKAGES}/libcppyy_backend.so
```
Comment (smuzaffar) [inline pip/py2-hepdata-lib.file]: @mrodozov , please set this in root.spec and drop these extra py[2,3]-[hepdata.rootpy]  files.

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
test parameters:
- full_cmssw = true

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
abort

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
@etejedor 
the latest trouble we have integrating the new pyroot:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5d0e5f/4797/unitTests.log
search for string import-rootpy
```
===== Test "import-rootpy" ====
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/py2-rootpy/1.0.1-5d0e5f/lib/python2.7/site-packages/rootpy/__init__.py", line 31, in <module>
    from .utils import quickroot as QROOT
  File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/lcg/root/6.21.01-5d0e5f/lib/python2.7/site-packages/ROOT/_facade.py", line 102, in _importhook
    return _orig_ihook(name, *args, **kwds)
  File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/py2-rootpy/1.0.1-5d0e5f/lib/python2.7/site-packages/rootpy/utils/quickroot.py", line 15, in <module>
    root_module = ROOT.module._root
AttributeError: 'module' object has no attribute '_root'

---> test import-rootpy had ERRORS
```

suggests the pip package https://github.com/rootpy/rootpy need to be adjusted, please advise on how to do that, or should I open an issue on the pyroot repo

**Comment by etejedor** [comment]
Hi @mrodozov, from what I see they were using this internal data member called `_root`, which we don't have in new PyROOT. By looking at the old PyROOT code, it seems this data member pointed to the libPyROOT extension module (`libPyROOT.so`). The equivalent in the new PyROOT is the libcppyy extension module (`libcppyy.so`), so they would need to do `import libcppyy` and use it instead of using the internal member `ROOT.module._root`.

Note though that since this is a new cppyy, some of the functions/variables they were using from the extension module might have changed name, but we can discuss this in a second iteration if necessary.

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
@silviodonato when the tests are finished I'd rather merge this PR and continue from there with 4 broken unit tests (see the links above)

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals
lets get this in IBs to run all tests. rootpy unit tests failures are due to the fact that it does not support new pyroot.

**Pull request merged**